src/arch/x86/pagetable_walker.cc

```diff
@@ -584,7 +597,10 @@ Walker::WalkerState::setupWalk(Addr vaddr)
     entry.vaddr = vaddr;
 
     Request::Flags flags = Request::PHYSICAL;
-    if (cr3.pcd)
+
+    // PCD can't be used if CR4.PCIDE=1 [sec 2.5
+    // of Intel's Software Developer's manual]
+    if (cr3.pcd && cr4.pcide!=1)
```

Comment:
Just for readability, I'd do the following:

```suggestion
    if (!cr4.pcide && cr3.pcd)
```

This reads, if we're not using PCID, then check the cr3.pcd :).